### PR TITLE
libtorrent: bump openssl + modernize more for conan v2

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by Croydon — DISMISSED** [review]

**Review by jwillikers — DISMISSED** [review]

**Review by jwillikers — DISMISSED** [review]

**Comment by SpaceIm** [comment]
```
======== Testing the package: Executing test ========
libtorrent/1.2.3 (test package): Running test()
libtorrent/1.2.3 (test package): RUN: ./test_package magnet:?xt=urn:btih:c12fe1c06bba254a9dc9f519b335aa7c1367a88a
dyld: Symbol not found: _iconv
  Referenced from: /usr/lib/libarchive.2.dylib
  Expected in: /Users/jenkins/w/prod-v2/BuildSingleReference/p/libic50650dc30c43d/p/lib/libiconv.2.dylib
 in /usr/lib/libarchive.2.dylib
/bin/sh: line 1: 72984 Abort trap: 6           ./test_package magnet:?xt=urn:btih:c12fe1c06bba254a9dc9f519b335aa7c1367a88a

ERROR: libtorrent/1.2.3 (test package): Error in test() method, line 28
	self.run(f"{bin_path} {magnet_url}", env="conanrun")
	ConanException: Error 134 while executing
```

WTF. What is linked to libarchive in this dependency graph?

**Review by jwillikers — APPROVED** [review]

File: recipes/libtorrent/all/conanfile.py [hunk 1]
```diff
@@ -62,65 +81,55 @@ def configure(self):
...
+        if Version(self.version) < "2.0.0" or Version(self.version) >= "2.0.7":
+            self.requires("boost/1.81.0", transitive_headers=True)
         else:
-            self.requires("boost/1.76.0")
+            self.requires("boost/1.76.0", transitive_headers=True)
         if self.options.enable_encryption:
-            self.requires("openssl/1.1.1s")
+            self.requires("openssl/1.1.1t", transitive_headers=True, transitive_libs=True)
         if self.options.enable_iconv:
             self.requires("libiconv/1.17")
```
Comment (prince-chrismc) [inline recipes/libtorrent/all/conanfile.py:94]: Should this have Transitive libs for v2?

Comment (SpaceIm) [reply recipes/libtorrent/all/conanfile.py:94]: I don't know. Do you have a proof that openssl headers are included by libtorrent public headers?

Comment (prince-chrismc) [reply recipes/libtorrent/all/conanfile.py:94]: libtorrent/1.2.3 (test package): RUN: ./test_package magnet:?xt=urn:btih:c12fe1c06bba254a9dc9f519b335aa7c1367a88a
dyld: Symbol not found: _iconv
  Referenced from: /usr/lib/libarchive.2.dylib
  Expected in: /Users/jenkins/w/prod-v2/BuildSingleReference/p/libic50650dc30c43d/p/lib/libiconv.2.dylib
 in /usr/lib/libarchive.2.dylib
/bin/sh: line 1: 30073 Abort trap: 6           ./test_package magnet:?xt=urn:btih:c12fe1c06bba254a9dc9f519b335aa7c1367a88a

The v2 failure above says it needed the libs?

Comment (SpaceIm) [reply recipes/libtorrent/all/conanfile.py:94]: Why a runtime loader error related to libarchive and system iconv would be related to transitive_headers & transitive_libs of openssl which are compile & link time considerations (moreover of a different lib)?

Comment (prince-chrismc) [reply recipes/libtorrent/all/conanfile.py:94]: I didn't comment on the openssl line 🤔 I comment on a blank line lol I meant to comment on line 93 not 94

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**